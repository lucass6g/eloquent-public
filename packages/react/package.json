{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
