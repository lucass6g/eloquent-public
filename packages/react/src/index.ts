export {Switch, Case, Default} from './components/conditional-switch'