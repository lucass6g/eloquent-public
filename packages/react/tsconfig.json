{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
