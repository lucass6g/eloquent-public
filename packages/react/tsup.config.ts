import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})