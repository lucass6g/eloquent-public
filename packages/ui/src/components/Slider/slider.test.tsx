import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { Slider } from "./";
import React from "react";

describe("Slider Component", () => {
  it("renders with default value when no value is provided", () => {
    const component = render(<Slider data-testid="slider-test" />);
    const sliderElement = screen.getByRole("slider");

    expect(sliderElement?.getAttribute("aria-valuenow")).toBe("0");
    expect(sliderElement?.getAttribute("aria-valuemin")).toBe("0");
    expect(sliderElement?.getAttribute("aria-valuemax")).toBe("100");
    expect(component.baseElement).toMatchSnapshot();
  });

  it("renders with provided value", () => {
    const component = render(
      <Slider
        data-testid="slider-test"
        max={99}
        min={1}
        step={1}
        defaultValue={[10]}
      />
    );
    const sliderElement = screen.getByRole("slider");

    expect(sliderElement?.getAttribute("aria-valuenow")).toBe("10");
    expect(sliderElement?.getAttribute("aria-valuemin")).toBe("1");
    expect(sliderElement?.getAttribute("aria-valuemax")).toBe("99");
    expect(component.baseElement).toMatchSnapshot();
  });
});
