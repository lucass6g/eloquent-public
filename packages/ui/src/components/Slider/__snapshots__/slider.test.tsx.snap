// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Slider Component > renders with default value when no value is provided 1`] = `
<body>
  <div>
    <span
      aria-disabled="false"
      class="relative flex w-full touch-none select-none items-center"
      data-orientation="horizontal"
      data-testid="slider-test"
      dir="ltr"
      style="--radix-slider-thumb-transform: translateX(-50%);"
    >
      <span
        class="relative h-1.5 w-full grow overflow-hidden rounded-full bg-primary/20"
        data-orientation="horizontal"
      >
        <span
          class="absolute h-full bg-primary"
          data-orientation="horizontal"
          style="left: 0%; right: 100%;"
        />
      </span>
      <span
        style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
      >
        <span
          aria-orientation="horizontal"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
          class="block h-4 w-4 rounded-full border border-primary/50 bg-background shadow transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50"
          data-orientation="horizontal"
          data-radix-collection-item=""
          role="slider"
          style=""
          tabindex="0"
        />
      </span>
    </span>
  </div>
</body>
`;

exports[`Slider Component > renders with provided value 1`] = `
<body>
  <div>
    <span
      aria-disabled="false"
      class="relative flex w-full touch-none select-none items-center"
      data-orientation="horizontal"
      data-testid="slider-test"
      dir="ltr"
      style="--radix-slider-thumb-transform: translateX(-50%);"
    >
      <span
        class="relative h-1.5 w-full grow overflow-hidden rounded-full bg-primary/20"
        data-orientation="horizontal"
      >
        <span
          class="absolute h-full bg-primary"
          data-orientation="horizontal"
          style="left: 0%; right: 90.81632653061224%;"
        />
      </span>
      <span
        style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(9.183673469387756% + 0px);"
      >
        <span
          aria-orientation="horizontal"
          aria-valuemax="99"
          aria-valuemin="1"
          aria-valuenow="10"
          class="block h-4 w-4 rounded-full border border-primary/50 bg-background shadow transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50"
          data-orientation="horizontal"
          data-radix-collection-item=""
          role="slider"
          style=""
          tabindex="0"
        />
      </span>
    </span>
  </div>
</body>
`;
